backend/bom-bom-server/src/main/java/me/bombom/api/v1/common/resolver/LoginMemberArgumentResolver.java

```diff
@@ -0,0 +1,44 @@
+package me.bombom.api.v1.common.resolver;
+
+import me.bombom.api.v1.auth.dto.CustomOAuth2User;
+import me.bombom.api.v1.common.exception.ErrorDetail;
+import me.bombom.api.v1.common.exception.UnauthorizedException;
+import me.bombom.api.v1.member.domain.Member;
+import org.springframework.core.MethodParameter;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+public class LoginMemberArgumentResolver implements HandlerMethodArgumentResolver {
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        return parameter.hasParameterAnnotation(LoginMember.class)
+                && parameter.getParameterType().equals(Member.class);
+    }
+
+    @Override
+    public Object resolveArgument(
+            MethodParameter parameter,
+            ModelAndViewContainer mavContainer,
+            NativeWebRequest webRequest,
+            WebDataBinderFactory binderFactory
+    ) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null ||  !authentication.isAuthenticated()) {
+            throw new UnauthorizedException(ErrorDetail.UNAUTHORIZED);
+        }
+        Object principal = authentication.getPrincipal();
+        if (principal instanceof CustomOAuth2User) {
+            Member member = ((CustomOAuth2User) principal).getMember();
+            if (member == null) {
+                throw new UnauthorizedException(ErrorDetail.UNAUTHORIZED);
+            }
+            return member;
+        }
+        throw new  UnauthorizedException(ErrorDetail.INVALID_TOKEN);
+    }
+}
```

Comment:
p4 

이렇게 하면 가독성이 조금 더 좋아질 것 같은데 어떠신가용? 😊

```suggestion
    @Override
    public Object resolveArgument(
            MethodParameter parameter,
            ModelAndViewContainer mavContainer,
            NativeWebRequest webRequest,
            WebDataBinderFactory binderFactory
    ) {
    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
    Member member = extractMember(auth);
    return member;
}

private Member extractMember(Authentication auth) {
    if (auth == null || !auth.isAuthenticated()) {
        throw new UnauthorizedException(ErrorDetail.UNAUTHORIZED);
    }

    Object principal = auth.getPrincipal();
    if (!(principal instanceof CustomOAuth2User)) {
        throw new UnauthorizedException(ErrorDetail.INVALID_TOKEN);
    }

    Member member = ((CustomOAuth2User) principal).getMember();
    if (member == null) {
        throw new UnauthorizedException(ErrorDetail.UNAUTHORIZED);
    }
    return member;
}
```